core/lib/types/src/api/mod.rs

```diff
@@ -728,6 +733,22 @@ pub enum BlockStatus {
     Verified,
 }
 
+/// Result tracers need to have a nested result field for compatibility. So we have two different
+/// structs 1 for blocks tracing and one for txs and call tracing
```

Comment:
This seems overly complicated. Why can't we always have `Vec<DebugCall>`?

Reply:
For compatibility reasons... i don't want to break our clients